--- Makefile-orig       2004-05-16 16:51:04.000000000 +0000
+++ Makefile    2005-11-09 17:37:34.000000000 +0000
@@ -20,7 +20,7 @@

 all:   default

-default:       libcgi-util.so libcgi-util.a test-cgis
+default:       libcgi-util.a test-cgis
        @echo
        @echo "cgi-util make options:"
        @echo "----------------------"
@@ -76,7 +76,7 @@
 cgi-util.o:    cgi-util.c cgi-util.h

 libcgi-util.so:
-       $(CC) $(CFLAGS) -fPIC cgi-util.c -c -o libcgi-util.o
+       $(CC) $(CFLAGS) cgi-util.c -c -o libcgi-util.o
        $(CC) -shared libcgi-util.o -o libcgi-util.so

 libcgi-util.a: cgi-util.o
