--- src/libFLAC/cpu.c.orig	2013-04-06 22:28:04.817282115 +0200
+++ src/libFLAC/cpu.c	2013-04-06 22:28:06.885292371 +0200
@@ -250,7 +250,7 @@
 #else
 			struct sigaction sigill_sse;
 			sigill_sse.sa_sigaction = sigill_handler_sse_os;
-			__sigemptyset(&sigill_sse.sa_mask);
+			sigemptyset(&sigill_sse.sa_mask);
 			sigill_sse.sa_flags = SA_SIGINFO | SA_RESETHAND; /* SA_RESETHAND just in case our SIGILL return jump breaks, so we don't get stuck in a loop */
 			if(0 == sigaction(SIGILL, &sigill_sse, &sigill_save))
 #endif
