diff --git a/gio/gcredentials.c b/gio/gcredentials.c
index 2827c45..067ca43 100644
--- a/gio/gcredentials.c
+++ b/gio/gcredentials.c
@@ -22,16 +22,11 @@
 
 #include "config.h"
 
-#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__OpenBSD__)
 #include <sys/types.h>
 #include <sys/socket.h>
-#include <string.h>
 #endif
-#ifdef __OpenBSD__
-#include <sys/types.h>
-#include <sys/socket.h>
 #include <string.h>
-#endif
 #include <stdlib.h>
 
 #include <gobject/gvaluecollector.h>
diff --git a/gio/gnetworkmonitornetlink.c b/gio/gnetworkmonitornetlink.c
index a276c58..2b0e211 100644
--- a/gio/gnetworkmonitornetlink.c
+++ b/gio/gnetworkmonitornetlink.c
@@ -22,6 +22,7 @@
 
 #include <errno.h>
 #include <unistd.h>
+#include <string.h>
 
 #include "gnetworkmonitornetlink.h"
 #include "gcredentials.h"
