--- src/mouse/lnxinpu2.c.orig	2013-04-04 15:21:47.625263298 +0200
+++ src/mouse/lnxinpu2.c	2013-04-04 15:21:50.581277944 +0200
@@ -33,7 +33,7 @@
  **/
 
 #include <stdio.h>
-#include <termio.h>
+#include <termios.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <fcntl.h>
@@ -56,8 +56,8 @@
 /*
  * keyboard stuff
  */
-static struct termio kbd_setup;
-static struct termio kbd_reset;
+static struct termios kbd_setup;
+static struct termios kbd_reset;
 static int kbd_initted = FALSE;
 static int kbd_enabled = TRUE;
 static int kbd_isatty;
