--- util/gifinto.c.orig	2013-04-12 13:58:06 +0000
+++ util/gifinto.c	2013-04-12 13:58:19 +0000
@@ -105,7 +105,7 @@
     if ( (strlen(FoutTmpName) + strlen(DEFAULT_TMP_NAME))  > STRLEN-1 ) GIF_EXIT("Filename too long.");
     strcat(FoutTmpName, DEFAULT_TMP_NAME);
     int FD;
-    FD = mkstemp(FoutTmpName); /* returns filedescriptor */
+    FD = _open(_mktemp(FoutTmpName), _O_CREAT | _O_EXCL | _O_TEMPORARY);
     if (FD == -1 )
     {
 	GIF_EXIT("Failed to open output.");
--- Makefile.am.orig	2013-04-24 18:44:01.410234279 +0200
+++ Makefile.am	2013-04-24 18:44:30.850380258 +0200
@@ -1,9 +1,4 @@
-SUBDIRS = lib util pic
-
-# Avoid trouble if your system doesn't have xmlto.
-if BUILD_DOC
-SUBDIRS += doc
-endif
+SUBDIRS = lib
 
 EXTRA_DIST = Makefile.unx \
 	     autogen.sh \
@@ -18,25 +13,3 @@
 	     $(WINDOWS)
 
 TESTS=test.sh
-
-# Auditing tools.
-
-# Coverity - avoid making doc as xmlto craps out under cov-build
-coverity:
-	cd lib; make
-	cd util; make
-
-# cppcheck should run clean
-cppcheck:
-	cppcheck -Ilib --inline-suppr --template gcc --enable=all --suppress=unusedStructMember --suppress=unusedFunction --force lib/*.[ch] util/*.[ch]
-
-# splint is nowhere near running clean
-SPLINT_SUPPRESSIONS = -boolops -nullret
-splint: 
-	splint +quiet +posixlib $(SPLINT_SUPPRESSIONS) lib/*.[ch]
-
-# release using the shipper tool, if distcheck passes
-release: distcheck
-	cd doc; make website
-	shipper -u -m -t; make clean
-	rm -fr doc/staging
