--- lib/gif_hash.c.orig	2013-04-27 19:31:08.351502669 +0200
+++ lib/gif_hash.c	2013-04-27 19:31:13.807529721 +0200
@@ -21,6 +21,7 @@
 #include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 #include "gif_lib.h"
 #include "gif_hash.h"
--- lib/dgif_lib.c.orig	2013-04-27 21:55:48 +0000
+++ lib/dgif_lib.c	2013-04-27 21:55:49 +0000
@@ -107,7 +107,7 @@
         free((char *)GifFile);
         return NULL;
     }
-#if defined(__MSDOS__) || defined(WINDOWS32) || defined(_OPEN_BINARY)
+#if defined(__MSDOS__) || defined(WINDOWS32) || defined(_OPEN_BINARY) || defined(_WIN32)
     setmode(FileHandle, O_BINARY);    /* Make sure it is in binary mode. */
 #endif /* __MSDOS__ */
 
--- lib/Makefile.in.orig	2013-04-27 21:56:22 +0000
+++ lib/Makefile.in	2013-04-27 21:56:32 +0000
@@ -236,7 +236,7 @@
                     gif_err.c   \
 					gif_lib_private.h
 
-libgif_la_LDFLAGS = -version-info @SHLIBVERSION@
+libgif_la_LDFLAGS = -no-undefined -version-info @SHLIBVERSION@
 libgif_la_LIBADD = @DEVS@
 libgif_CFLAGS = $(X_CFLAGS) $(AM_CFLAGS)
 all: all-am
--- Makefile.in.orig	2013-04-27 19:00:15 +0000
+++ Makefile.in	2013-04-27 19:00:18 +0000
@@ -229,7 +229,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = lib util doc pic
+SUBDIRS = lib
 EXTRA_DIST = Makefile.unx \
 	     autogen.sh \
 	     tests \
@@ -756,26 +756,6 @@
 	mostlyclean mostlyclean-generic mostlyclean-libtool pdf pdf-am \
 	ps ps-am tags tags-recursive uninstall uninstall-am
 
-
-# Auditing tools.
-
-# Coverity - avoid making doc as xmlto craps out under cov-build
-coverity:
-	cd lib; make
-	cd util; make
-
-# cppcheck should run clean
-cppcheck:
-	cppcheck -Ilib --inline-suppr --template gcc --enable=all --suppress=unusedStructMember --suppress=unusedFunction --force lib/*.[ch] util/*.[ch]
-splint: 
-	splint +quiet +posixlib $(SPLINT_SUPPRESSIONS) lib/*.[ch]
-
-# release using the shipper tool
-release: dist
-	cd doc; make website
-	shipper -u -m -t; make clean
-	rm -fr doc/staging
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
