--- util/gifinto.c.orig	2013-04-12 13:58:06 +0000
+++ util/gifinto.c	2013-04-12 13:58:19 +0000
@@ -105,7 +105,7 @@
     if ( (strlen(FoutTmpName) + strlen(DEFAULT_TMP_NAME))  > STRLEN-1 ) GIF_EXIT("Filename too long.");
     strcat(FoutTmpName, DEFAULT_TMP_NAME);
     int FD;
-    FD = mkstemp(FoutTmpName); /* returns filedescriptor */
+    FD = _open(_mktemp(FoutTmpName), _O_CREAT | _O_EXCL | _O_TEMPORARY);
     if (FD == -1 )
     {
 	GIF_EXIT("Failed to open output.");
