--- config.sub.orig	2011-04-03 15:49:04.097242005 +0200
+++ config.sub	2011-04-03 15:50:06.201241926 +0200
@@ -1191,7 +1191,7 @@
 		;;
 	xbox)
 		basic_machine=i686-pc
-		os=-mingw32
+		os=-xbox
 		;;
 	xps | xps100)
 		basic_machine=xps100-honeywell
@@ -1337,7 +1337,7 @@
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
-	      | -mingw32* | -linux-gnu* | -linux-android* \
+	      | -mingw32* | -xbox* | -linux-gnu* | -linux-android* \
 	      | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
