--- sqlite-3.2.7\configure	2005-09-17 11:34:54.000000000 +0000
+++ sqlite-3.2.7-mod\configure	2005-09-30 01:02:00.000000000 +0000
@@ -19594,9 +19594,9 @@
 fi
 if test "$TARGET_EXEEXT" = ".exe"; then
   OS_UNIX=0
-  OS_WIN=1
+  OS_WIN=0
   tclsubdir=win
-  TARGET_CFLAGS="$TARGET_CFLAGS -DOS_WIN=1"
+  TARGET_CFLAGS="$TARGET_CFLAGS"
 else
   OS_UNIX=1
   OS_WIN=0

--- sqlite-3.2.7\configure.ac	2005-09-17 11:34:54.000000000 +0000
+++ sqlite-3.2.7-mod\configure.ac	2005-09-30 01:02:22.000000000 +0000
@@ -368,9 +368,9 @@
 fi
 if test "$TARGET_EXEEXT" = ".exe"; then
   OS_UNIX=0
-  OS_WIN=1
+  OS_WIN=0
   tclsubdir=win
-  TARGET_CFLAGS="$TARGET_CFLAGS -DOS_WIN=1"
+  TARGET_CFLAGS="$TARGET_CFLAGS"
 else
   OS_UNIX=1
   OS_WIN=0

--- sqlite-3.2.7\src\os_unix.c	2005-09-22 10:43:40.000000000 +0000
+++ sqlite-3.2.7-mod\src\os_unix.c	2005-09-30 01:04:30.000000000 +0000
@@ -1322,6 +1322,11 @@
 */
 char *sqlite3OsFullPathname(const char *zRelative){
   char *zFull = 0;
+#ifdef __MSDOS__
+  char zBuf[5000];
+  zBuf[0] = 0;
+  sqlite3SetString(&zFull, _truename( zRelative, zBuf ), (char*)0);
+#else
   if( zRelative[0]=='/' ){
     sqlite3SetString(&zFull, zRelative, (char*)0);
   }else{
@@ -1334,6 +1339,7 @@
                     (char*)0);
     sqliteFree(zBuf);
   }
+#endif
   return zFull;
 }
 
