--- configure.in.orig	Wed Apr 20 02:36:56 2005
+++ configure.in	Sun Apr  8 22:43:37 2007
@@ -49,6 +49,25 @@
 		AC_PATH_PROG(STRIP,$host_alias-strip)
     ;;
 
+  *-*-mingw32*)
+    #
+    # As we are not cross compiling we keep the AC_PROG_PATH that is currently defined for all utils
+    # 
+		AC_PATH_PROG(AR,ar.exe)
+		AC_PATH_PROG(AS,as.exe)
+		AC_PATH_PROG(LD,ld.exe)
+		AC_PATH_PROG(NM,nm.exe)
+		AC_PATH_PROG(GCC,gcc.exe)
+		AC_PATH_PROG(CXX,g++.exe)
+		AC_PATH_PROG(RANLIB,ranlib.exe)
+		AC_PATH_PROG(DLLTOOL,dlltool.exe)
+		AC_PATH_PROG(WINDRES,windres.exe)
+		AC_PATH_PROG(STRIP,strip.exe)
+		
+		AC_SUBST(host_alias,$host)
+		AC_SUBST(XDKEXEEXT,".exe")
+    ;;
+
   *)
     AC_MSG_ERROR(Error: Build environment "$build" is currently not supported.)
     ;;
