--- src-IL/src/il_png.c.orig	2013-04-04 18:21:31.299914134 +0200
+++ src-IL/src/il_png.c	2013-04-04 18:21:46.447989250 +0200
@@ -278,7 +278,7 @@
 
 	// Expand low-bit-depth grayscale images to 8 bits
 	if (png_color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) {
-		png_set_gray_1_2_4_to_8(png_ptr);
+		png_set_expand_gray_1_2_4_to_8(png_ptr);
 	}
 
 	// Expand RGB images with transparency to full alpha channels
--- src-IL/src/il_icon.c.orig	2013-04-04 18:24:34.744823773 +0200
+++ src-IL/src/il_icon.c	2013-04-04 18:25:06.364980593 +0200
@@ -525,7 +525,7 @@
 
 	// Expand low-bit-depth grayscale images to 8 bits
 	if (ico_color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) {
-		png_set_gray_1_2_4_to_8(ico_png_ptr);
+		png_set_expand_gray_1_2_4_to_8(ico_png_ptr);
 	}
 
 	// Expand RGB images with transparency to full alpha channels
--- src-ILU/ilur/ilur.c.orig	2013-04-04 18:29:21.546245962 +0200
+++ src-ILU/ilur/ilur.c	2013-04-04 18:29:35.350314409 +0200
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
 #include <malloc.h>
