--- progs/demos/smooth_opengl3/smooth_opengl3.c.orig	2013-03-25 17:14:35.010868007 +0100
+++ progs/demos/smooth_opengl3/smooth_opengl3.c	2013-03-25 17:14:37.626880965 +0100
@@ -102,7 +102,7 @@
 typedef void (APIENTRY *PFNGLBINDBUFFERPROC) (GLenum target, GLuint buffer);
 typedef void (APIENTRY *PFNGLBUFFERDATAPROC) (GLenum target, ourGLsizeiptr size, const GLvoid *data, GLenum usage);
 typedef GLuint (APIENTRY *PFNGLCREATESHADERPROC) (GLenum type);
-typedef void (APIENTRY *PFNGLSHADERSOURCEPROC) (GLuint shader, GLsizei count, const ourGLchar **string, const GLint *length);
+typedef void (APIENTRY *PFNGLSHADERSOURCEPROC) (GLuint shader, GLsizei count, const ourGLchar * const *string, const GLint *length);
 typedef void (APIENTRY *PFNGLCOMPILESHADERPROC) (GLuint shader);
 typedef GLuint (APIENTRY *PFNGLCREATEPROGRAMPROC) (void);
 typedef void (APIENTRY *PFNGLATTACHSHADERPROC) (GLuint program, GLuint shader);
