--- src/gd_errors.h.orig	2013-04-24 20:45:57.822715302 +0200
+++ src/gd_errors.h	2013-04-24 20:45:59.710724664 +0200
@@ -1,7 +1,7 @@
 #ifndef GD_ERRORS_H
 #define GD_ERRORS_H
 
-#ifndef WIN32
+#if !defined WIN32 && !defined _WIN32 && !defined __DJGPP__
 # include <syslog.h>
 #else
 /*
--- src/gd_bmp.c.orig	2013-04-24 20:46:47.174960025 +0200
+++ src/gd_bmp.c	2013-04-24 20:46:48.798968082 +0200
@@ -792,7 +792,7 @@
 	}
 
 	/* The line must be divisible by 4, else its padded with NULLs */
-	padding = ((int)ceill(0.1 * info->width)) % 4;
+	padding = ((int)__builtin_ceill(0.1 * info->width)) % 4;
 	if (padding) {
 		padding = 4 - padding;
 	}
