--- src/libjasper/base/jas_getopt.c.orig	2013-04-04 17:53:46.479658735 +0200
+++ src/libjasper/base/jas_getopt.c	2013-04-04 17:54:03.279742043 +0200
@@ -76,6 +76,7 @@
 
 #include "jasper/jas_getopt.h"
 #include "jasper/jas_math.h"
+#include "jasper/jas_debug.h"
 
 /******************************************************************************\
 * Global data.
--- src/libjasper/bmp/bmp_dec.c.orig	2013-04-04 17:55:18.068112899 +0200
+++ src/libjasper/bmp/bmp_dec.c	2013-04-04 17:55:35.132197519 +0200
@@ -77,6 +77,7 @@
 #include "jasper/jas_stream.h"
 #include "jasper/jas_image.h"
 #include "jasper/jas_malloc.h"
+#include "jasper/jas_debug.h"
 
 #include "bmp_cod.h"
 
--- src/libjasper/jpc/jpc_t1dec.c.orig	2007-01-19 22:43:07.000000000 +0100
+++ src/libjasper/jpc/jpc_t1dec.c	2013-04-04 17:57:47.620854491 +0200
@@ -78,6 +78,7 @@
 #include "jasper/jas_fix.h"
 #include "jasper/jas_stream.h"
 #include "jasper/jas_math.h"
+#include "jasper/jas_debug.h"
 
 #include "jpc_bs.h"
 #include "jpc_mqdec.h"
--- src/libjasper/jpc/jpc_tsfb.h.orig	2013-04-04 18:00:31.945669340 +0200
+++ src/libjasper/jpc/jpc_tsfb.h	2013-04-04 18:01:44.534029278 +0200
@@ -126,9 +126,13 @@
 
 /* Perform analysis. */
 int jpc_tsfb_analyze(jpc_tsfb_t *tsfb, jas_seq2d_t *x);
+int jpc_tsfb_analyze2(jpc_tsfb_t *tsfb, int *a, int xstart, int ystart,
+  int width, int height, int stride, int numlvls);
 
 /* Perform synthesis. */
 int jpc_tsfb_synthesize(jpc_tsfb_t *tsfb, jas_seq2d_t *x);
+int jpc_tsfb_synthesize2(jpc_tsfb_t *tsfb, int *a, int xstart, int ystart,
+  int width, int height, int stride, int numlvls);
 
 /* Get band information for a TSFB. */
 int jpc_tsfb_getbands(jpc_tsfb_t *tsfb, uint_fast32_t xstart,
--- src/libjasper/mif/mif_cod.c.orig	2013-04-04 18:03:46.502634088 +0200
+++ src/libjasper/mif/mif_cod.c	2013-04-04 18:04:21.446807368 +0200
@@ -70,6 +70,7 @@
 #include "jasper/jas_image.h"
 #include "jasper/jas_string.h"
 #include "jasper/jas_malloc.h"
+#include "jasper/jas_debug.h"
 
 #include "mif_cod.h"
 
--- src/libjasper/pnm/pnm_dec.c.orig	2013-04-04 18:06:09.207341725 +0200
+++ src/libjasper/pnm/pnm_dec.c	2013-04-04 18:06:31.415451847 +0200
@@ -79,6 +79,7 @@
 #include "jasper/jas_types.h"
 #include "jasper/jas_stream.h"
 #include "jasper/jas_image.h"
+#include "jasper/jas_debug.h"
 
 #include "pnm_cod.h"
 
