--- src/appl/tmrdemo.c.orig	2013-04-10 13:03:18 +0000
+++ src/appl/tmrdemo.c	2013-04-10 13:03:32 +0000
@@ -1,4 +1,5 @@
 #include <jasper/jasper.h>
+#include <windows.h>
 
 int main(int argc, char **argv)
 {
@@ -43,7 +44,7 @@
 	printf("zero time %.3f us\n", t * 1e6);
 
 	jas_tmr_start(&tmr);
-	sleep(1);
+	Sleep(1000);
 	jas_tmr_stop(&tmr);
 	t = jas_tmr_get(&tmr);
 	printf("time delay %.8f s\n", t);
