--- jmorecfg.h.orig	2013-04-04 16:48:16.384170443 +0200
+++ jmorecfg.h	2013-04-04 16:49:03.684405003 +0200
@@ -252,15 +252,15 @@
  * Defining HAVE_BOOLEAN before including jpeglib.h should make it work.
  */
 
-#ifdef HAVE_BOOLEAN
 #ifndef FALSE			/* in case these macros already exist */
 #define FALSE	0		/* values of boolean */
 #endif
 #ifndef TRUE
 #define TRUE	1
 #endif
-#else
-typedef enum { FALSE = 0, TRUE = 1 } boolean;
+
+#ifndef HAVE_BOOLEAN
+typedef enum { booleanFALSE = 0, booleanTRUE = 1 } boolean;
 #endif
 
 
