--- Project/GNU/Library/configure.ac.orig	2013-02-19 11:42:14.000000000 +0100
+++ Project/GNU/Library/configure.ac	2013-04-06 03:54:00.250261483 +0200
@@ -684,8 +684,8 @@
 dnl ### MediaInfoLib flags
 dnl #########################################################################
 
-MediaInfoLib_LIBS=" -L$(pwd) -l$with_libmediainfo_name $MediaInfoLib_LIBS"
-MediaInfoLib_LIBS_Static=" $(pwd)/.libs/lib$with_libmediainfo_name.a $MediaInfoLib_LIBS_Static"
+MediaInfoLib_LIBS=" -l$with_libmediainfo_name $MediaInfoLib_LIBS"
+MediaInfoLib_LIBS_Static=" -l$with_libmediainfo_name $MediaInfoLib_LIBS_Static"
 
 dnl #########################################################################
 dnl ### Output
--- Project/GNU/Library/libmediainfo.pc.in.orig	2012-12-15 10:37:24.000000000 +0100
+++ Project/GNU/Library/libmediainfo.pc.in	2013-04-06 03:54:25.690387622 +0200
@@ -3,7 +3,7 @@
 libdir=@libdir@
 includedir=@includedir@
 Unicode=@MediaInfoLib_Unicode@
-Libs_Static=@libdir@/lib@MediaInfoLib_LibName@.a ${libdir}/libzen.a -lpthread -lz
+Libs_Static=-L@libdir@ -l@MediaInfoLib_LibName@ -lzen -lz -lpthread
 la_name=lib@MediaInfoLib_LibName@.la
 
 Name: libmediainfo
