--- src/linux_sysfs.c.orig	2013-03-23 20:56:22.154629952 +0100
+++ src/linux_sysfs.c	2013-03-23 20:56:24.406641116 +0100
@@ -44,6 +44,7 @@
 #include <sys/mman.h>
 #include <dirent.h>
 #include <errno.h>
+#include <linux/limits.h> /* for PATH_MAX */
 
 #if defined(__i386__) || defined(__x86_64__) || defined(__arm__)
 #include <sys/io.h>
--- src/pciaccess_private.h.orig	2013-03-23 20:51:07.613070129 +0100
+++ src/pciaccess_private.h	2013-03-23 20:51:13.609099939 +0100
@@ -42,7 +42,7 @@
  * know of any OS we support where this isn't available in a sufficiently
  * new version, so warn unconditionally.
  */
-#include <sys/fcntl.h>
+#include <fcntl.h>
 
 #ifndef O_CLOEXEC
 #warning O_CLOEXEC not available, please upgrade.
