--- lib/zipint.h.orig	2013-05-13 16:21:19.037145199 +0200
+++ lib/zipint.h	2013-05-13 16:21:41.449256329 +0200
@@ -40,7 +40,6 @@
 #include <zlib.h>
 
 #ifdef _WIN32
-#define ZIP_EXTERN __declspec(dllexport)
 /* for dup(), close(), etc. */
 #include <io.h>
 #endif
--- lib/zip.h.orig	2013-05-13 16:20:48.780995163 +0200
+++ lib/zip.h	2013-05-13 16:21:58.313339961 +0200
@@ -36,15 +36,7 @@
 
 
 
-#ifndef ZIP_EXTERN
-#ifdef _WIN32
-#define ZIP_EXTERN __declspec(dllimport)
-#elif defined(__GNUC__) && __GNUC__ >= 4
-#define ZIP_EXTERN __attribute__ ((visibility ("default")))
-#else
 #define ZIP_EXTERN
-#endif
-#endif
 
 #ifdef __cplusplus
 extern "C" {
