--- Project/GNU/Library/configure.ac.orig	2012-07-08 09:28:59.000000000 +0200
+++ Project/GNU/Library/configure.ac	2013-04-06 03:38:15.561577023 +0200
@@ -122,8 +122,8 @@
 dnl -------------------------------------------------------------------------
 dnl Common
 dnl
-ZenLib_LIBS="-L$(pwd) -lzen -lpthread -lstdc++ $ZenLib_LIBS"
-ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a -lpthread -lstdc++ $ZenLib_LIBS_Static"
+ZenLib_LIBS="-lzen -lpthread -lstdc++ $ZenLib_LIBS"
+ZenLib_LIBS_Static="-lzen -lpthread -lstdc++ $ZenLib_LIBS_Static"
 
 dnl #########################################################################
 dnl ### C/C++ compiler options used to compile
--- Project/GNU/Library/libzen.pc.in.orig	2012-07-08 09:29:11.000000000 +0200
+++ Project/GNU/Library/libzen.pc.in	2013-04-06 03:42:26.818822929 +0200
@@ -4,7 +4,7 @@
 includedir=@includedir@
 Unicode=@ZenLib_Unicode@
 WstringMissing=@ZenLib_wstring_missing@
-Libs_Static=@libdir@/libzen.a -lpthread
+Libs_Static=-L@libdir@ -lzen -lpthread
 
 Name: libzen
 Version: @PACKAGE_VERSION@
--- Project/GNU/Library/Makefile.am.orig	2013-04-05 21:25:42.504829830 +0200
+++ Project/GNU/Library/Makefile.am	2013-04-05 21:25:44.104837756 +0200
@@ -31,6 +31,7 @@
 zenincludedir = $(includedir)/ZenLib
 zeninclude_HEADERS = \
  ../../../Source/ZenLib/BitStream.h \
+ ../../../Source/ZenLib/BitStream_Fast.h \
  ../../../Source/ZenLib/BitStream_LE.h \
  ../../../Source/ZenLib/Conf.h \
  ../../../Source/ZenLib/Conf_Internal.h \
